lib/screens/post_slot_screen.dart

```diff
@@ -0,0 +1,219 @@
+          ', ' +
+          placemarks[0].locality;
+    });
+  }
+
+  postSlot() {
+    // TODO: Add validation
+    CollectionReference slotsDb =
+        FirebaseFirestore.instance.collection('slots');
+    slotsDb.doc(uuid).set({
+      'title': titleController.text,
+      'latitude': selectedPosition.latitude,
+      'longitude': selectedPosition.longitude,
+      'daily': dailyController.text,
+      'hourly': hourlyController.text,
+      'imageUrl': slotImage,
+      'userUid': user.uid,
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    StorageReference imgStorage = FirebaseStorage.instance.ref();
+    if (position == null) {
+      return Material(child: Center(child: Text("loading...")));
+    } else {
+      return Scaffold(
+        appBar: AppBar(
+          centerTitle: true,
+          title: Text('Add new slot'),
+        ),
+        body: Column(
+          children: <Widget>[
+            MediaQuery.of(context).viewInsets.bottom > 1
+                ? Container()
+                : Container(
+                    height: MediaQuery.of(context).size.height * 0.3,
+                    child: GoogleMap(
+                      markers: Set.from(markers),
+                      initialCameraPosition: CameraPosition(
+                        target: LatLng(position.latitude, position.longitude),
+                        zoom: 14,
+                      ),
+                      onTap: (position) {
+                        selectedPosition = Position(
+                          latitude: position.latitude,
+                          longitude: position.longitude,
+                        );
+                        setMarker();
+                      },
+                    ),
+                  ),
+            Expanded(
+              child: Padding(
+                padding: EdgeInsets.all(15),
+                child: ListView(
+                  children: <Widget>[
+                    Row(
+                      children: <Widget>[
+                        GestureDetector(
+                          onTap: () async {
+                            PickedFile pickedImage = await ImagePicker()
+                                .getImage(source: ImageSource.gallery);
+                            String imageId =
+                                selectedPosition.latitude.toString() +
+                                    '_' +
+                                    position.longitude.toString();
+                            StorageUploadTask uploadTask = imgStorage
+                                .child(imageId)
+                                .putFile(File(pickedImage.path));
+
+                            String url = await (await uploadTask.onComplete)
+                                .ref
+                                .getDownloadURL();
+                            setState(() {
+                              slotImage = url;
+                            });
+                          },
```

Comment:
This inline function could be a good method of your class.

Reply:
There will be some refactoring and function outsourcing after the last issue for this week.
This is not the only function that should be outsourced and there are some widgets I have to extract too.